arrow-core/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -220,6 +220,18 @@ fun <A, B, C> Either<A, B>.flatMap(f: (B) -> Either<A, C>): Either<A, C> = fold(
  */
 inline fun <B> Either<*, B>.getOrElse(crossinline default: () -> B): B = fold({ default() }, { it })
 
+/**
+ * Returns the value from this [Either.Right] or the the given argument if this is a [Either.Left] while providing access to
+ * the value of [Either.Left].
+ *
+ * Example:
+ * ```
+ * Right(12).getOrElseGet { 17 } // Result: 12
+ * Left(12).getOrElseGet { it + 5 } // Result: 17
+ * ```
+ */
+inline fun <A, B> Either<A, B>.getOrElseGet(crossinline default: (A) -> B): B = fold({ default(it) }, { it })
```

Comment:
Since We have already in `ApplicativeError` `handleError` which is a polymorphic version of this method I think a better name would be just `handle` since `handleError` in this context would be inaccurate given `A` is not always used to represent an error.

Reply:
Another name that could work here is just `getOrHandle`

Reply:
The idea of this name comes from the functional Java lib vavr. I agree that an Either.left is not always representing an error. I could live with „getOrHandle“, „getOrResolve“ or maybe simply „resolve“.

Reply:
If handleError exists, let's reuse that name right?

Reply:
@pakoito I would prefer `getOrHandle` as it doesnt imply that `Left` is an error.

Reply:
@raulraja Changed to `getOrHandle`.